### PR TITLE
Added documentation around stable branch

### PR DESCRIPTION
**Pull request by clacroix12**
Fixes: #3898 

Signed-off-by: Coady LaCroix <clacroix@redhat.com>

### CODE REVIEW

File: docs/stable_branch.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+## Overview
+The OCS-CI team has adopted the use of a stable branch for ocs-ci. The intention
+here is to perform additional testing on code changes that are merged to our master branch
+before they are deemed stable. Once testing has passed, the new changes will be propagated
+to our stable branch and can be used for production testing or any other testing that
+requires the use of our stable branch.
+
+The initial proposal, discussion, and implementation details can be found in a
+[shared google doc](https://docs.google.com/document/d/15rqGVDEWFTFsg6FOOBeWUYYrwBa9wi0rZtdfOYi_-zg/edit).
```
Comment (petr-balogh) [inline docs/stable_branch.md:11]: Make a comment that it's available only for RH internal accounts

Comment (clacroix12) [reply docs/stable_branch.md:11]: Done

File: docs/stable_branch.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  in our production testing.
+
+## Automation
+Testing new changes to master branch and qualifying them for promotion to the stable
+branch is done through scheduled automation. The current schedule is to run tests twice
+weekly and automatically promote changes that pass testing. Automation is done via a
+[jenkins pipeline](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-ocs-ci-stable-branch-pipeline/).
+
+If all tests pass, a fast-forward merge is performed to update the stable branch with
+the latest changes to master. An email is also sent out stating that the promotion was
```
Comment (petr-balogh) [inline docs/stable_branch.md:43]: I would also mention that the new tag is created, e.g.  stable-ocs-4.7-202103021622 https://github.com/red-hat-storage/ocs-ci/tags

Comment (clacroix12) [reply docs/stable_branch.md]: done!

**Review by RazTamir — APPROVED** [review]

**Review by vavuthu — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Review by mbukatov — APPROVED** [review]

**Pull request merged**